include/exadg/aero_acoustic/calculators/source_term_calculator.h

```diff
@@ -87,6 +95,25 @@ class SourceTermCalculator
 
     Number rho = static_cast<Number>(data.density);
 
+    // In case we blend in the source term we check, if the scaling is space dependent. Only in that
+    // case we have to evaluate the function in every equadrature point. Otherwise the scaling
+    // purely temporal and constant during this function.
+    if(data.blend_in)
+      AssertThrow(data.blend_in_function != nullptr,
+                  dealii::ExcMessage("No blend-in function provided."));
+
+    bool const space_dependent_scaling =
+      data.blend_in_function != nullptr ? data.blend_in_function->varies_in_space() : false;
+    Number const pure_temporal_scaling_factor =
+      data.blend_in_function != nullptr ? data.blend_in_function->compute_time_factor() : 1.0;
+
+    auto apply_scaling = [&](auto & flux, auto const & q) {
+      if(space_dependent_scaling)
+        flux *= FunctionEvaluator<0, dim, Number>::value(*data.blend_in_function, q, time);
+      else
+        flux *= pure_temporal_scaling_factor;
```

Comment:
It took me some time to understand the complete logic behind this code.

I suggest to write

```C++
Number const pure_temporal_scaling_factor =
     (not space_dependent_scaling) ? data.blend_in_function->compute_time_factor() : 1.0;
```

It think this would be somewhat more intuitive and I think this would make clear that the variable `pure_temporal_scaling_factor` is expected to be used only for one particular case.

Reply:
[f763625](https://github.com/exadg/exadg/pull/644/commits/f763625ed2c96df3c3b3e917c4c1085b58158e44)